src/map/mob.c

```diff
@@ -3646,77 +3646,354 @@ void item_dropratio_adjust(int nameid, int mob_id, int *rate_adjust)
+	}
+	if (idx == MAX_MOB_DROP && libconfig->setting_get_elem(t, i)) {
+		ShowWarning("Too many drops in mob %d\n", class_);
+	}
+}
+
 /*==========================================
  * processes one mobdb entry
  *------------------------------------------*/
-bool mob_parse_dbrow(char** str) {
-	struct mob_db *db, entry;
+bool mob_read_db_sub(config_setting_t *mobt, int id, const char *source)
+{
+	struct mob_db *entry = NULL, tmpEntry;
+	config_setting_t *t = NULL;
+	int i32 = 0, class_ = 0;
 	struct status_data *mstatus;
-	int class_, i;
-	double exp, maxhp;
 	struct mob_data data;
+	const char *str = NULL;
+	double maxhp;
+	double exp;
 
-	class_ = atoi(str[0]);
+	entry = &tmpEntry;
+	if (!libconfig->setting_lookup_int(mobt, "Id", &class_)) {
+		ShowWarning("mob_read_db_sub: Missing id in \"%s\", entry #%d, skipping.\n", source, class_);
+		return false;
+	}
 
 	if (class_ <= 1000 || class_ > MAX_MOB_DB) {
-		ShowError("mob_parse_dbrow: Invalid monster ID %d, must be in range %d-%d.\n", class_, 1000, MAX_MOB_DB);
+		ShowError("mob_read_db_sub: Invalid monster ID %d, must be in range %d-%d.\n", class_, 1000, MAX_MOB_DB);
 		return false;
 	}
 	if (pc->db_checkid(class_)) {
-		ShowError("mob_parse_dbrow: Invalid monster ID %d, reserved for player classes.\n", class_);
+		ShowError("mob_read_db_sub: Invalid monster ID %d, reserved for player classes.\n", class_);
 		return false;
 	}
 
 	if (class_ >= MOB_CLONE_START && class_ < MOB_CLONE_END) {
-		ShowError("mob_parse_dbrow: Invalid monster ID %d. Range %d-%d is reserved for player clones. Please increase MAX_MOB_DB (%d).\n", class_, MOB_CLONE_START, MOB_CLONE_END-1, MAX_MOB_DB);
+		ShowError("mob_read_db_sub: Invalid monster ID %d. Range %d-%d is reserved for player clones. Please increase MAX_MOB_DB (%d).\n", class_, MOB_CLONE_START, MOB_CLONE_END-1, MAX_MOB_DB);
 		return false;
 	}
 
-	memset(&entry, 0, sizeof(entry));
+	memset(&tmpEntry, 0, sizeof(tmpEntry));
 
-	db = &entry;
-	mstatus = &db->status;
+	mstatus = &entry->status;
+
+	entry->vd.class_ = class_;
+
+	if (!libconfig->setting_lookup_string(mobt, "SpriteName", &str) || !*str ) {
+		ShowWarning("mob_read_db_sub: Missing SpriteName in mob %d of \"%s\", skipping.\n", class_, source);
+		aFree(entry);
+		return false;
+	} else {
+		safestrncpy(entry->sprite, str, sizeof(entry->sprite));
+	}
 
-	db->vd.class_ = class_;
-	safestrncpy(db->sprite, str[1], sizeof(db->sprite));
-	safestrncpy(db->jname, str[2], sizeof(db->jname));
-	safestrncpy(db->name, str[3], sizeof(db->name));
-	db->lv = atoi(str[4]);
-	db->lv = cap_value(db->lv, 1, USHRT_MAX);
-	mstatus->max_hp = atoi(str[5]);
-	mstatus->max_sp = atoi(str[6]);
+	if (!libconfig->setting_lookup_string(mobt, "Name", &str) || !*str ) {
+		ShowWarning("mob_read_db_sub: Missing Name in mob %d of \"%s\", skipping.\n", class_, source);
+		aFree(entry);
+		return false;
+	} else {
+		safestrncpy(entry->name, str, sizeof(entry->sprite));
+		safestrncpy(entry->jname, str, sizeof(entry->sprite));
```

Comment:
Is there a reason why we want to keep name and jname? If not, we can completely remove one of the two